bdb/summarize.c

```diff
@@ -361,11 +361,9 @@ int bdb_summarize_table(bdb_state_type *bdb_state, int ixnum, int comp_pct,
         goto done;
     }
 
-#if defined(_IBM_SOURCE) || defined(__linux__)
+#if defined(POSIX_FADV_SEQUENTIAL)
     // inform kernel that we will be accessing file sequentially
-    // and that we won't need the file to be cached
-    fdatasync(fd);
```

Comment:
Analyze doesn't care if dirty pages in the kernel page cache have been sync'd to the disk device. So no need to fsync here.

Reply:
I think the fdatasync() was done so advise would be applied; not sure if needed nowadays.
https://insights.oetiker.ch/linux/fadvise/ seems to use it for that purpose too.

Reply:
It's only *recommended* when using `POSIX_FADV_DONTNEED`. Like I said, there's no reason for analyze to care if data has been sync'd to device. So it's an overkill to fsync here.

Reply:
Even with `O_DIRECT`, `fsync()` will likely still have a performance impact here. From `fsync` man page:

> This includes writing through or flushing a disk cache if present.

`O_DIRECT` bypasses kernel page cache, however the backing device may have its own cache. According to the man page, `fsync()` will flush the device cache too.